File: 0537-complex-number-multiplication/0537-complex-number-multiplication.py
class Solution:
    def complexNumberMultiply(self, num1: str, num2: str) -> str:
        
        fR,fC = num1.split("+")
        fR = int(fR)
        fC = int(fC.split("i")[0])
        
        sR,sC = num2.split("+")
        sR = int(sR)
        sC = int(sC.split("i")[0])
        
        real1 = (fR*sR)
        mixed1 = (fR*sC)
        mixed2 = (fC*sR)
        complex1 = (fC*sC)
        
        real_ = real1 - complex1
        complex_ = mixed1 + mixed2
        return "%d+%di"%(real_,complex_)        
        

File: 0537-complex-number-multiplication/0537-complex-number-multiplication.cpp
class Solution {
public:
    string complexNumberMultiply(string num1, string num2) {
        int fR = 0, fC = 0, sR = 0, sC = 0;
        sscanf(num1.c_str(), "%d+%di", &fR,&fC);
        sscanf(num2.c_str(), "%d+%di", &sR,&sC);
        return to_string(fR*sR-fC*sC)+"+"+to_string(fR*sC+fC*sR)+"i";
    }
};